api_server.py

```
@@ -31,9 +43,34 @@
 )
 logger = logging.getLogger(__name__)
 
-# Store extracted cases
+# Store extracted cases (in-memory fallback)
 EXTRACTED_CASES: Dict[str, Any] = {}
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)